pkg/controller/build/image_build_request.go

```diff
@@ -43,82 +43,70 @@ type ImageInfo struct {
 
 // Represents the request to build a layered OS image.
 type ImageBuildRequest struct {
-	// The target MachineConfigPool
-	Pool *mcfgv1.MachineConfigPool
-	// The base OS image (derived from the machine-config-osimageurl ConfigMap)
-	BaseImage ImageInfo
-	// The extensions image (derived from the machine-config-osimageurl ConfigMap)
-	ExtensionsImage ImageInfo
-	// The final OS image (desired from the on-cluster-build-config ConfigMap)
-	FinalImage ImageInfo
-	// The OpenShift release version (derived from the machine-config-osimageurl ConfigMap)
-	ReleaseVersion string
-	// An optional user-supplied Dockerfile that gets injected into the build.
-	CustomDockerfile string
-}
-
-type buildInputs struct {
-	onClusterBuildConfig *corev1.ConfigMap
-	osImageURL           *corev1.ConfigMap
-	customDockerfiles    *corev1.ConfigMap
-	pool                 *mcfgv1.MachineConfigPool
-	machineConfig        *mcfgv1.MachineConfig
+	// The target Build object
+	MachineOSBuild *mcfgv1alpha1.MachineOSBuild
+	// the cofig the build is based off of
+	MachineOSConfig *mcfgv1alpha1.MachineOSConfig
+	// the containerfile designated from the MachineOSConfig
+	Containerfile string
 }
 
 // Constructs a simple ImageBuildRequest.
-func newImageBuildRequest(pool *mcfgv1.MachineConfigPool) ImageBuildRequest {
-	return ImageBuildRequest{
-		Pool: pool.DeepCopy(),
+func newImageBuildRequest(mosc *mcfgv1alpha1.MachineOSConfig, mosb *mcfgv1alpha1.MachineOSBuild) *ImageBuildRequest {
+	ibr := &ImageBuildRequest{
+		MachineOSConfig: mosc,
+		MachineOSBuild:  mosb,
 	}
-}
 
-// Populates the final image info from the on-cluster-build-config ConfigMap.
-func newFinalImageInfo(inputs *buildInputs) ImageInfo {
-	return ImageInfo{
-		Pullspec: inputs.onClusterBuildConfig.Data[FinalImagePullspecConfigKey],
-		PullSecret: corev1.LocalObjectReference{
-			Name: inputs.onClusterBuildConfig.Data[FinalImagePushSecretNameConfigKey],
-		},
+	// only support noArch for now
+	for _, file := range mosc.Spec.BuildInputs.Containerfile {
+		if file.ContainerfileArch == mcfgv1alpha1.NoArch {
```

Comment:
note: Unless `ContainerfileArch` is set, a build will never get started. It would be great if we could move that validation to the API to ensure that at least one architecture is set and not accept the object vs. let it get this far.